docker-storage-setup.sh

```diff
@@ -865,15 +990,74 @@ setup_storage() {
    Fatal "Storage is already configured with ${current_driver} driver. Can't configure it with ${STORAGE_DRIVER} driver. To override, remove $DOCKER_STORAGE and retry."
   fi
 
-  # Set up lvm thin pool LV
+  # If a user decides to setup both (a) and (b):
+  # a) lvm thin pool for devicemapper.
+  # b) a separate volume for docker root.
+  # (a) will have a higher priority than (b) and will be setup first.
+
+  # Set up lvm thin pool LV.
   if [ "$STORAGE_DRIVER" == "devicemapper" ]; then
     setup_lvm_thin_pool
   else
       write_storage_config_file
   fi
+
+  # If docker root is on a separate volume, setup that.
+  if ! setup_docker_root_lv_fs; then
+    Error "Failed to setup docker root volume."
+    return 1
+  fi
+}
+
+restore_selinux_context() {
+  local dir=$1
+
+  if ! restorecon -R $dir; then
+    Error "restorecon -R $dir failed."
+    return 1
+  fi
+}
+
+get_docker_root_dir(){
+    local flag=false
+    options=$(grep -e "^OPTIONS" /etc/sysconfig/docker|cut -d"'" -f 2)
+    for opt in $options
+    do
+        if [ "$flag" = true ];then
+           DOCKER_ROOT_DIR=$opt
+           flag=false
+           continue
+        fi
+	case "$opt" in
+            "-g"|"--graph")
+                flag=true
+                ;;
```

Comment:
Does this handle the case of two or more spaces -g and value. For example `-g    /var/lib/foo`